Signal/src/view controllers/CountryCodeViewController.m

```diff
@@ -20,17 +24,61 @@ - (void)viewDidLoad {
     [self.navigationController.navigationBar setTranslucent:NO];
     _countryCodes = [self countryCodesForSearchTerm:nil];
     self.title    = NSLocalizedString(@"COUNTRYCODE_SELECT_TITLE", @"");
+    self.searchBar.delegate = self;
 }
 
 - (NSArray *)countryCodesForSearchTerm:(NSString *)searchTerm {
     NSMutableArray *result = [NSMutableArray array];
-    for (NSString *countryCode in [PhoneNumberUtil countryCodesForSearchTerm:nil]) {
+    for (NSString *countryCode in [PhoneNumberUtil countryCodesForSearchTerm:searchTerm]) {
         NSString *callingCode = [self callingCodeFromCountryCode:countryCode];
         if (callingCode != nil &&
             ![callingCode isEqualToString:@"+0"]) {
             [result addObject:countryCode];
         }
     }
+    
+    if (searchTerm.length > 0) {
+        NSArray *allCountryCodes = [self countryCodesForSearchTerm:nil];
+        // We want to ignore + when using a search term like "+44".
+        // We also want to ignore punctuation, etc.
+        // To keep things simple, we just remove everything except letters and numbers.
+        NSCharacterSet *charactersToRemove = [[NSMutableCharacterSet alphanumericCharacterSet] invertedSet];
+        NSString *simplifiedSearchTerm = [[searchTerm componentsSeparatedByCharactersInSet:charactersToRemove]
+                                          componentsJoinedByString:@"" ];
+        for (NSString *split in [simplifiedSearchTerm componentsSeparatedByCharactersInSet:[NSCharacterSet whitespaceCharacterSet]]) {
```

Comment:
Am I confused or is this PR out of date?  

Now that the search logic is in SSK, it seems like we shouldn't need to do any of this here.